File: 2491-divide-players-into-teams-of-equal-skill/2491-divide-players-into-teams-of-equal-skill.java
class Solution {
    public long dividePlayers(int[] skill) {
        Arrays.sort(skill);
        long ref = skill[0] + skill[skill.length - 1];
        long result = skill[0] * skill[skill.length - 1];
        int lo = 1, hi = skill.length - 2;

        while (lo < hi) {
            long chk = skill[lo] + skill[hi];
            if (chk != ref) return -1;
            result += skill[lo] * skill[hi];
            lo++;
            hi--;
        }

        return result;
    }
}

File: 2491-divide-players-into-teams-of-equal-skill/2491-divide-players-into-teams-of-equal-skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()
        ref = skill[0] + skill[-1]
        lo, hi = 1, len(skill) - 2
        result = skill[0] * skill[-1]

        while lo < hi:
            chk = skill[lo] + skill[hi]
            if chk != ref:
                return -1
            result += skill[lo] * skill[hi]
            lo += 1
            hi -= 1

        return result